controllers/component/component_adapter.go

```diff
@@ -81,6 +100,10 @@ func (a *Adapter) EnsureComponentIsCleanedUp() (controller.OperationResult, erro
 	}
 
 	if len(snapshotComponents) == 0 {
+		err = h.RemoveFinalizerFromComponent(a.client, a.logger, a.context, a.component, h.ComponentFinalizer)
```

Comment:
I think we should remove the finalizer from the component whether `len(snapshotComponents) == 0` or not here,  to make the component be able to be deleted, so we can combine this line and L117 by move this line out of thif `if`, how do you think of it?

Reply:
Yes I agree the code could be cleaned up @hongweiliu17 what are your thoughts on this refactor [here](https://github.com/redhat-appstudio/integration-service/pull/410/files#diff-0a268eab99b22ffc9ca9a74ef2752ed7dd05a822b3da8fe160ffee3a2ad1035eR102) . I think the prior code should have made a call to StopProcessing where it would Continue which I also fixed here.

Reply:
Agree. Thanks.

Reply:
I like the refactoring here. Based on the latest changes, does this mean that now we'll continue to reconcile a Component that has been marked for deletion and doesn't have a finalizer on it (so it will soon get deleted by K8s)? Wouldn't it cause a problem in the future if some other Ensure* function tries to process this component?

Reply:
I'm just thinking if removing `return controller.StopProcessing()` was the right decision, what do you think?

Reply:
I think this is similar to the way the PLR controller handles reconciliation around finalizer removal but maybe I am misunderstanding when it's correct to StopProcessing vs other options

Reply:
I won't fight for it because the possibility of getting into a problematic state is low for now.
After all, we'd need to have:
* an additional Ensure* function that gets called right after this one (`EnsureComponentIsCleanedUp`)
* a Component (without finalizer, with deletionTimestamp), which is about to be deleted in 0.000001 secs, that gets picked up for reconciliation by this new Ensure* function.